spec.md

```diff
@@ -143,10 +167,18 @@ message CheckAccessRequest {
 }
 
 message CheckAccessResponse {
+  enum Result {
+    RESULT_ALLOWED = 0;
+    RESULT_DENIED = 1;
+  }
+
+  Result result = 1;
 }
 ```
 
-`CheckAccess` is a REQUIRED operation which checks that a subject has access to perform the given actions on the given resources. If all actions are allowed, runtime implementations MUST respond with a `CheckAccessResponse`. If any action is not allowed, runtime implementations MUST respond with gRPC status 7 (PERMISSION_DENIED) and an appropriate error message. As with the Validation API, if the subject credential is not valid, runtime implementations MUST respond with gRPC status 16 (UNAUTHENTICATED) and an appropriate error message.
+`CheckAccess` is a REQUIRED operation which checks that the subject identified by the given credential has access to perform the given actions on the given resources. If all given actions are allowed, runtime implementations MUST respond with `result` set to `RESULT_ALLOWED`. If any action is not allowed, runtime implementations MUST respond with `result` set to `RESULT_DENIED`.
```

Comment:
I know we've talked about this - should this include an optional "reason"?

Reply:
I'm not sure what a reason for denial would look like right now - in practice both SpiceDB (which we use currently in Infratographer) and OpenFGA just return allow/deny responses. There's not really a free text "reason" we could bubble up from that besides some nicely formatted "x does not have access to resource y" message, which I would expect to be handled by a runtime implementation.